pkg/server/registry/apigroups/acorn/apps/confirmupgrade.go

```diff
@@ -35,7 +37,19 @@ func (s *ConfirmUpgradeStrategy) Create(ctx context.Context, obj types.Object) (
 	// The app validation logic should not run there.
 	app := &v1.AppInstance{}
 	err := s.client.Get(ctx, kclient.ObjectKey{Namespace: ri.Namespace, Name: ri.Name}, app)
-	if err != nil {
+	if err != nil && apierrors.IsNotFound(err) {
```

Comment:
nit, `err != nil` not needed.
```suggestion
	if apierrors.IsNotFound(err) {
```